homeassistant/components/switchbee/cover.py

```diff
@@ -166,9 +166,9 @@ async def async_set_cover_position(self, **kwargs: Any) -> None:
             raise HomeAssistantError(
                 f"Failed to set {self._attr_name} position to {str(kwargs[ATTR_POSITION])}, error: {str(exp)}"
             ) from exp
-        else:
-            cast(
-                SwitchBeeShutter, self.coordinator.data[self._device.id]
-            ).position = kwargs[ATTR_POSITION]
-            self.coordinator.async_set_updated_data(self.coordinator.data)
-            self.async_write_ha_state()
+
+        cast(
+            SwitchBeeShutter, self.coordinator.data[self._device.id]
+        ).position = kwargs[ATTR_POSITION]
```

Comment:
You can create this helper in the base `SwitchBeeDeviceEntity`:
```python
    def _get_coordinator_device(self) -> _DeviceTypeT:
        return cast(_DeviceTypeT, self.coordinator.data[self._device.id])
```

And replace this with:
```python
    self._get_coordinator_device().position = kwargs[ATTR_POSITION]
```